src/commands/apm_cmds/publish.js

```diff
@@ -251,24 +277,35 @@ exports.task = function ({
             }
 
             const getMajor = version => version.split('.')[0]
+
             ctx.isMajor = getMajor(repo.version) != getMajor(ctx.version)
           }
         }
       ])
     },
     {
-      title: 'Compile contracts',
-      task: async () => (await compileContracts()),
-      enabled: () => !onlyContent && web3Utils.isAddress(contract)
-    },
-    {
-      title: 'Deploy contract',
-      task: async (ctx) => {
-        const deployTaskParams = { contract, init, reporter, network, cwd, web3, apmOptions }
-        
-        return await deploy.task(deployTaskParams)
+      title: 'Compile and deploy contracts',
+      skip: (ctx, task) => {
+        if (!ctx.isMajor || onlyContent) {
+          return 'Publishing patch or minor version, skipping contract compilation and deployment.'
+        }
       },
-      enabled: ctx => !onlyContent && ((contract && !web3Utils.isAddress(contract)) || (!contract && ctx.isMajor && !reuse) || automaticallyBump)
+      task: ctx => new TaskList([
+        {
+          title: 'Compile contracts',
+          task: async () => (await compileContracts()),
+          enabled: () => !onlyContent && web3Utils.isAddress(contract)
+        },
+        {
+          title: 'Deploy contract',
+          task: async (ctx) => {
+            const deployTaskParams = { contract, init, reporter, network, cwd, web3, apmOptions }
+
+            return await deploy.task(deployTaskParams)
+          },
+          enabled: ctx => !onlyContent && ((contract && !web3Utils.isAddress(contract)) || (!contract && ctx.isMajor && !reuse) || automaticallyBump)
```

Comment:
Can be simplified to `!web3Utils.isAddress(contract) && !reuse` 🎉

Reply:
👍